File: 8vim/src/main/java/inc/flide/vim8/utils/DialogsHelper.java
package inc.flide.vim8.utils;

import android.content.Context;
import com.afollestad.materialdialogs.MaterialDialog;
import com.afollestad.materialdialogs.list.DialogSingleChoiceExtKt;
import inc.flide.vim8.R;
import java.util.ArrayList;
import java.util.Collection;
import kotlin.Unit;
import kotlin.jvm.functions.Function1;
import kotlin.jvm.functions.Function3;

public final class DialogsHelper {
    private DialogsHelper() {

    }

    public static MaterialDialog createMaterialDialog(Context context, int titleRes, int messageRes,
                                                      int positiveButtonRes,
                                                      Function1<MaterialDialog, Unit> callback) {
        return new MaterialDialog(context, MaterialDialog.getDEFAULT_BEHAVIOR()).title(titleRes, null)
                .message(messageRes, null, null).cancelable(false)
                .cancelOnTouchOutside(false)
                .positiveButton(positiveButtonRes, null, callback)
                .negativeButton(R.string.activate_ime_dialog_negative_button_text, null, null);
    }

    public static MaterialDialog createItemsChoice(Context context, int titleRes,
                                                   Collection<String> items, int selectedIndex,
                                                   Function3<MaterialDialog, Integer, CharSequence, Unit> callback) {
        return DialogSingleChoiceExtKt.listItemsSingleChoice(
                new MaterialDialog(context, MaterialDialog.getDEFAULT_BEHAVIOR()).title(titleRes, null)
                        .positiveButton(R.string.generic_okay_text, null, null)
                        .negativeButton(R.string.generic_cancel_text, null, null), null,
                new ArrayList<>(items), null, selectedIndex, true, -1, -1, callback);
    }
}

File: 8vim/src/main/kotlin/inc/flide/vim8/utils/DialogsHelper.kt
package inc.flide.vim8.utils

import android.content.Context
import androidx.appcompat.app.AlertDialog
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import inc.flide.vim8.R
import java.util.function.IntConsumer

object DialogsHelper {
    fun createItemsChoice(
        context: Context,
        titleRes: Int,
        items: Collection<String>,
        selectedIndex: Int,
        callback: IntConsumer
    ): AlertDialog {
        var currentIndex = selectedIndex
        return MaterialAlertDialogBuilder(context)
            .setTitle(titleRes)
            .setPositiveButton(R.string.generic_okay_text) { _, _ ->
                if (currentIndex != -1 && currentIndex != selectedIndex) {
                    callback.accept(currentIndex)
                }
            }
            .setNegativeButton(R.string.generic_cancel_text, null)
            .setSingleChoiceItems(items.toTypedArray(), selectedIndex) { _, which ->
                currentIndex = which
            }
            .show()
    }

    fun showAlert(context: Context, titleRes: Int, message: String) {
        MaterialAlertDialogBuilder(context)
            .setTitle(titleRes)
            .setMessage(message)
            .setPositiveButton(R.string.generic_okay_text, null)
            .show()
            .show()
    }
}
